src/hooks/useFocusTrap.ts

```diff
@@ -22,10 +22,12 @@ interface FocusTrapHookSettings {
   disabled?: boolean
 
   /**
-   * If true, when this focus trap is cleaned up, restore focus to the element that had
-   * focus immediately before the focus trap was enabled. (Default: false)
+   * Return focus to the specified element when the focus trap is cleaned up or disabled.
+   * If true, return focus to the element that was focused immediately before the focus
+   * trap was enabled. If a function is passed, it will be called when the focus trap is
+   * cleaned up to determine the element to return focus to. (Default: false)
    */
-  restoreFocusOnCleanUp?: boolean
+  restoreFocusOnCleanUp?: HTMLElement | boolean | (() => HTMLElement | boolean)
```

Comment:
The new `useAwaitableState` seems to be all that is needed to fix the issue that brought on this PR.  @colebemis @smockle, do you think we should keep the changes for restoring focus as well, or revert back to the simpler setup and only use `useAwaitableState` to solve the immediate issue.